pkg/artifactory/resource_artifactory_push_replication.go

```diff
@@ -228,20 +231,38 @@ func packPushReplication(pushReplication *GetPushReplication, d *schema.Resource
 	errors = setValue("enable_event_replication", pushReplication.EnableEventReplication)
 
 	if pushReplication.Replications != nil {
+
+		// Get replications from TF state
+		var tfReplications []interface{}
+		if v, ok := d.GetOkExists("replications"); ok {
+			tfReplications = v.([]interface{})
+		}
+
 		var replications []map[string]interface{}
-		for _, repo := range pushReplication.Replications {
+		for _, repl := range pushReplication.Replications {
 			replication := make(map[string]interface{})
 
-			replication["url"] = repo.URL
-			replication["socket_timeout_millis"] = repo.SocketTimeoutMillis
-			replication["username"] = repo.Username
-			replication["password"] = repo.Password
-			replication["enabled"] = repo.Enabled
-			replication["sync_deletes"] = repo.SyncDeletes
-			replication["sync_properties"] = repo.SyncProperties
-			replication["sync_statistics"] = repo.SyncStatistics
-			replication["path_prefix"] = repo.PathPrefix
-			replication["proxy"] = repo.ProxyRef
+			replication["url"] = repl.URL
+			replication["socket_timeout_millis"] = repl.SocketTimeoutMillis
+			replication["username"] = repl.Username
+
+			// find the matching replication from current state
+			tfReplicationIndex := slices.IndexFunc(tfReplications, func(r interface{}) bool {
+				return r.(map[string]interface{})["url"] == repl.URL
+			})
+			if tfReplicationIndex != -1 {
+				// set password from current state to avoid state drift
+				// from missing password in Artifactory API response
+				existingPassword := tfReplications[tfReplicationIndex].(map[string]interface{})["password"]
```

Comment:
Any reason for variable existingPassword?  Is there a way to do this assignment directly to replication["password"].

Reply:
Sure.